### PR TITLE
Listen to events so that encryption icon updates when status changes

### PR DESCRIPTION
**Pull request by uhoreg**
Fixes https://github.com/element-hq/element-web/issues/28164

## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] I have licensed the changes to Element by completing the [Contributor License Agreement (CLA)](https://cla-assistant.io/element-hq/element-web)

### CODE REVIEW

File: src/hooks/useEncryptionStatus.ts [hunk 1]
```diff
@@ -6,21 +6,41 @@ SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only
...
-                setE2eStatus(e2eStatus);
-            });
-        }
-    }, [client, room]);
+    const updateEncryptionStatus = useMemo(
+        () =>
+            throttle(
+                () => {
+                    if (client.getCrypto()) {
+                        console.log("getting status");
```
Comment (t3chguy) [inline src/hooks/useEncryptionStatus.ts:24]: this is not a helpful console message

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

File: src/hooks/useEncryptionStatus.ts [hunk 2]
```diff
@@ -6,21 +6,41 @@ SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only
...
+                        });
+                    }
+                },
+                250,
+                { leading: true, trailing: true },
+            ),
+        [client, room],
+    );
+
+    useEffect(updateEncryptionStatus, [updateEncryptionStatus]);
```
Comment (BillCarsonFr) [inline src/hooks/useEncryptionStatus.ts:36]: I am not familiar with that, the second arg are dependencies? So is this not a cyclic dependencies?

**Review by BillCarsonFr — COMMENTED** [review]
Can't say much about react stuffs :/
There are no test for membership changes?

File: src/hooks/useEncryptionStatus.ts (hunk 2 above)
Comment (t3chguy) [reply src/hooks/useEncryptionStatus.ts:36]: the dependencies specify when `useEffect` should be called. So this says call `updateEncryptionState` whenever the method changes, and the method is a useMemo with its own dependencies, so it changes when `[client, room]` (either) changes.

File: src/hooks/useEncryptionStatus.ts (hunk 1 above)
Comment (uhoreg) [reply src/hooks/useEncryptionStatus.ts:24]: ugh.  I thought I removed all my debugging messages, but I forgot that one.

**Comment by uhoreg** [comment]
> There are no test for membership changes?

There is a test for membership changes at https://github.com/element-hq/element-web/pull/28407/files#diff-5463cabfb673ce908cb151cfb23ca1085a6a510c7d9f897c76379408ace43fe3R652-R671

**Review by BillCarsonFr — APPROVED** [review]
LGTM

**Pull request merged**